scripts/directtools/__init__.py

```diff
@@ -0,0 +1,450 @@
+from __future__ import (absolute_import, division, print_function)
+
+import collections
+from mantid import mtd
+from mantid.simpleapi import DeleteWorkspace, LineProfile, OneMinusExponentialCor, Transpose
+import matplotlib
+from matplotlib import pyplot
+import numpy
+from scipy import constants
+import time
+
+
+def _chooseMarker(markers, index):
+    """Pick a marker from markers and return next cyclic index."""
+    if index >= len(markers):
+        index = 0
+    marker = markers[index]
+    index += 1
+    return marker, index
+
+
+def _clearlatex(s):
+    """Return a string from which LaTeX formatting has been removed."""
+    s = s.translate(None, '$\\^{}')
+    s = s.replace('$', '')
+    s = s.replace('\\', '')
+    s = s.replace('^', '')
+    s = s.replace('{', '')
+    s = s.replace('}', '')
+    return s
+
+
+def _finalizeprofileE(axes):
+    """Set axes for const E axes."""
+    axes.set_xlim(xmin=0.)
+    axes.set_xlabel('$Q$ (\\AA$^{-1}$)')
+    axes.set_ylim(0.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto Q-range: {}...{} \xc5-1'.format(xMin, xMax))
+
+
+def _finalizeprofileQ(workspaces, axes):
+    """Set axes for const Q axes."""
+    workspaces = _normwslist(workspaces)
+    axes.set_xlim(xmin=-10.)
+    axes.set_xlabel('Energy (meV)')
+    cMax = 0.
+    for i in range(len(workspaces)):
+        c = numpy.nanmax(workspaces[0].readY(0))
+        if c > cMax:
+            cMax = c
+    axes.set_ylim(ymin=0., ymax=cMax / 100.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto E-range: {}...{} meV'.format(xMin, xMax))
+
+
+def _globalnanminmax(workspaces):
+    """Return global minimum and maximum of the given workspaces."""
+    workspaces = _normwslist(workspaces)
+    globalMin = numpy.inf
+    globalMax = -numpy.inf
+    for ws in workspaces:
+        cMin, cMax = nanminmax(ws)
+        if cMin < globalMin:
+            globalMin = cMin
+        if cMax > globalMax:
+            globalMax = cMax
+    return globalMin, globalMax
+
+
+def _label(ws, cut, width, singleWS, singleCut, singleWidth, quantity, units):
+    """Return a line label for a line profile."""
+    ws = _normws(ws)
+    logs = SampleLogs(ws)
```

Comment:
`logs` is redefined inside `if not singleWS:`: is it needed in the outer scope or can this line be removed?